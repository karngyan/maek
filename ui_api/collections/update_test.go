package collections_test

import (
	"context"
	"fmt"
	"testing"

	approvals "github.com/approvals/go-approval-tests"
	"github.com/karngyan/maek/domains/collections"
	"github.com/karngyan/maek/ui_api/testutil"
	"github.com/stretchr/testify/assert"
)

func TestUpdate(t *testing.T) {
	defer testutil.TruncateTables()
	cs := testutil.NewClientStateWithUser(t)

	c, err := collections.CreateCollection(context.Background(), cs.Workspace.ID, cs.User.ID)
	assert.NoError(t, err)

	rr, err := cs.Put(fmt.Sprintf("/v1/workspaces/%d/collections/%d", cs.Workspace.ID, c.ID), map[string]any{
		"name":        "Updated Name",
		"description": "Updated Description",
	})
	assert.Nil(t, err)

	assert.Equal(t, 200, rr.Code)
	approvals.VerifyJSONBytes(t, rr.Body.Bytes())
}
