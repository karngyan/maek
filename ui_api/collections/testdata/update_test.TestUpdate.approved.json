{
  "collection": {
    "created": 1234567890,
    "createdById": 1,
    "description": "Updated Description",
    "favorite": false,
    "id": 1,
    "name": "Updated Name",
    "trashed": false,
    "updated": 1234567890,
    "updatedById": 1,
    "workspaceId": 1
  }
}